File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    String getPalindrome(String s, int l, int r) {
        String res = "";
        while(l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            res = s.substring(l, r+1);
            l--;
            r++;
        }
        return res;
    }

    public String longestPalindrome(String s) {
        String res = "";
        for(int i=0; i<s.length(); i++) {
            String a = getPalindrome(s, i, i);
            String b = getPalindrome(s, i, i+1);
            String str = a.length() > b.length() ? a : b;
            if(str.length() > res.length()) {
                res = str;
            }
        }
        return res;
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.ts
function longestPalindrome(s: string): string {

    const getPalindromeLength = (l, r) => {
        while(l >= 0 && r < s.length && s.charAt(l) === s.charAt(r)) {
            l--
            r++
        }
        return s.slice(l+1,r)
    }

    let res = ""
    for(let i=0; i<s.length; i++) { //O(n)
        const str1 = getPalindromeLength(i,i)
        const str2 = getPalindromeLength(i,i+1)
        // console.log("i: ", i, " str1: ", str1, " str2: ", str2)
        if(str1.length > res.length) {
            res = str1
        }
        if(str2.length > res.length){
            res = str2
        }
    }
    return res
};